libs/core/langchain_core/language_models/chat_models.py

```diff
@@ -1100,6 +1102,7 @@ async def _agenerate_with_cache(
                 **result.generations[0].message.response_metadata,
             }
         if check_cache and llm_cache:
+            result.generations = loads(dumps(result.generations))
```

Comment:
Wouldn't the problem be in the caching implementation? Mutating .generations here does not seem like it can be a correct solution